app.js

```diff
@@ -3,11 +3,19 @@ const { getTopics } = require("./controllers/topics.controller");
 const app = express();
 
 //GET
-app.get("/api/topics", getTopics);
+app.get("/api/:endpoint", getTopics);
+
+//Error Handling 
+app.use((err, req, res, next) => {
+  if (err.code === "42601") {
```

Comment:
This error code is from a syntax error in a db query, we would not be expecting users to make syntax errors in database queries as they should never be interacting with the database. Your 404 errors should be from missing information, such as a valid search with no rows, or an endpoint that doesn't exist. PSQL error codes will traditionally be for searching with incorrect data types or missing keys.